ops/model.py

```diff
@@ -271,6 +277,7 @@ def __init__(self, name, backend, cache):
         self._cache = cache
         self._is_our_unit = self.name == self._backend.unit_name
         self._status = None
```

Comment:
Related to the prior comment, note how there's a bunch of data initialized here instead of passed into the constructor. Again, I'm not sure about this, so take it with a grain of salt, but it's worth looking into it and discussing it at least.